ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -116,370 +112,535 @@ def run_cmd(cmdline):
+                flows[in_port][out_port] = action
+        self.debug('Parsed flows on bridge:\n{}'.format(json.dumps(flows, indent=2)))
+
+        # Transform parsed flows to self.flows dict
+        for in_port in flows:
+            if self.sides[in_port] == NIC:
+                # From NIC to TORs, upstream flow
+                self.flows['upstream']['in_side'] = NIC
+                self.flows['upstream']['out_sides'] = [self.sides[out_port] for out_port, action in flows[in_port].items() if action == OUTPUT]
+            else:
+                # From TOR to NIC, downstream flow
+                self._init_active_standby(self.sides[in_port])
+                self.flows['downstream']['in_side'] = self.sides[in_port]
+                self.flows['downstream']['out_sides'] = [self.sides[out_port] for out_port, action in flows[in_port].items() if action == OUTPUT]
+
+    @property
+    def status(self):
+        """Property for statu of the mux bridge.
+
+        Status of the mux bridge is maintained in instance attributes. This property is to gather the attributes and
+        return them in a dict.
+        """
+        self.lock.acquire()
+
+        # Transform mux flows to json expected by mux simulator client
+        flows = {}
+        flows[self.ports[NIC]] = [
+            {'action': OUTPUT, 'out_port': self.ports[out_side]} for out_side in self.flows['upstream']['out_sides']
+        ]
+
+        if self.flows['downstream']['in_side'] is not None:
+            in_side = self.flows['downstream']['in_side']
+            in_port = self.ports[in_side]
+            flows[in_port] = [
+                {'action': OUTPUT, 'out_port': self.ports[out_side]} for out_side in self.flows['downstream']['out_sides']
+            ]
+
+        healthy = True
+        if len(self.flows['downstream']['out_sides']) != 1 or len(self.flows['upstream']['out_sides']) != 2:
+            healthy = False
+
+        status = {
+            'bridge': self.bridge,
+            'vm_set': self.vm_set,
+            'port_index': self.port_index,
+            'ports': self.ports,
+            'active_port': self.active_port,
+            'active_side': self.active_side,
+            'standby_side': self.standby_side,
+            'standby_port': self.standby_port,
+            'flows': flows,
+            'flap_counter': self.flap_counter,
+            'healthy': healthy
+        }
+        self.lock.release()
+        return status
+
+    def set_active_side(self, new_active_side):
+        """Set the active side of the mux bridge to the specified side.
+
+        If the specified side is same as the current active side of bridge, no config change is reuqired. Otherwise,
+        this method will run ovs-ofctl command to remove flow and add a new flow to switch active side. All the
+        related instance attributes are updated after open flow rules are changed.
+        """
+        try:
```

Comment:
The right semantics to use a lock is:
```python
some_lock.acquire()
try:
    # do something...
finally:
    some_lock.release()
```
Or it would be more Pythonic to wrap this logic in a `with` statement.

Reply:
Thanks for pointing this out! Updated.